### PR TITLE
gui: Show current wallet name in window title

### PR DESCRIPTION
**Pull request by promag**
screenshot 2019-01-11 at 23 58 26

### CODE REVIEW

File: src/qt/bitcoingui.cpp [hunk 1]
```diff
@@ -567,10 +567,9 @@ void BitcoinGUI::setClientModel(ClientModel *_clientModel)
 }
 
 #ifdef ENABLE_WALLET
-bool BitcoinGUI::addWallet(WalletModel *walletModel)
+void BitcoinGUI::addWallet(WalletModel *walletModel)
```
Comment (promag) [inline src/qt/bitcoingui.cpp]: nit, `WalletModel* walletModel`.

File: src/qt/bitcoingui.h [hunk 2]
```diff
@@ -80,8 +80,8 @@ class BitcoinGUI : public QMainWindow
         The wallet model represents a bitcoin wallet, and offers access to the list of transactions, address book and sending
         functionality.
     */
-    bool addWallet(WalletModel *walletModel);
-    bool removeWallet(WalletModel* walletModel);
+    void addWallet(WalletModel *walletModel);
```
Comment (promag) [inline src/qt/bitcoingui.h]: Same as above.

**Comment by jonasschnelli** [comment]
Yeah. Why not.
Concept ACK.
I suggest removing the text part "Wallet" | "Node".

**Comment by promag** [comment]
> I suggest removing the text part "Wallet" | "Node".

@jonasschnelli I kind of agree, not very helpful I guess.

**Comment by promag** [comment]
Now with Jonas suggestion.

**Comment by hebasto** [comment]
Concept ACK.

**Review by hebasto — CHANGES_REQUESTED** [review]

**Comment by luke-jr** [comment]
IMO this doesn't really make sense, since the same window is used for all loaded wallets currently.

**Comment by jonasschnelli** [comment]
> IMO this doesn't really make sense, since the same window is used for all loaded wallets currently.

IMHO, window titles are okay to be context sensitive.

File: src/qt/bitcoingui.cpp [hunk 3]
```diff
@@ -1208,6 +1201,19 @@ void BitcoinGUI::updateProxyIcon()
...
 }
 
+void BitcoinGUI::updateWindowTitle()
+{
+    QString window_title = tr(PACKAGE_NAME) + " - ";
+#ifdef ENABLE_WALLET
+    if (enableWallet) {
+        WalletView* const walletView = walletFrame ? walletFrame->currentWalletView() : nullptr;
+        if (walletView) window_title += walletView->getWalletModel()->getDisplayName() + " - ";
+    }
```
[the remark this answers is not in the document]
Comment (hebasto) [reply src/qt/bitcoingui.cpp]: @promag [from IRC](http://www.erisian.com.au/bitcoin-core-dev/log-2019-01-13.html#l-159)

> walletFrame can be empty right?

Exactly! That is why I suppose checking `walletFrame` is sufficient and checking `enableWallet` and `walletView` is redundant.

**Comment by hebasto** [comment]
@promag [from IRC](http://www.erisian.com.au/bitcoin-core-dev/log-2019-01-13.html#l-159)

> walletFrame can be empty right?

Exactly! That is why I suppose checking `walletFrame` is sufficient and checking `enableWallet` and `walletView` is redundant.

**Comment by promag** [comment]
@hebasto dropped `enableWallet` check, `walletFrame` is indeed sufficient:
https://github.com/bitcoin/bitcoin/blob/a9b71a09a0bbbdebc4c0b10d287bf2d53f628cf4/src/qt/bitcoingui.cpp#L101-L104 

Still checking `walletView` for the case the last wallet is unloaded.

Let me know so I can squash.

**Comment by hebasto** [comment]
tACK c1e8f74f326f862adbf5a326eb1b9a65e6eb1edd

Checked the case when all wallets are unloaded:
screenshot from 2019-01-14 10-38-00

**Comment by promag** [comment]
Thanks @hebasto.

**Comment by laanwj** [comment]
Concept ACK, agree it's fine for window titles to be dynamic and context sensitive.

**Comment by hebasto** [comment]
re-tACK b0b2d112550817e31e469a19a2d69bcef97ae4be

**Comment by Sjors** [comment]
Concept ACK, code changes look fine, tested.

I don't like the default behavior:  
schermafbeelding 2019-01-15 om 13 01 36

"- [default wallet]" is visual clutter. If there's only one wallet loaded and it doesn't have a custom name, I would just show nothing.

**Comment by promag** [comment]
Rebased after #14594 merge.

**Comment by Sjors** [comment]
> That would be the same as no wallet is loaded.

That shouldn't cause any confusion. Either a wallet is visible or not.

It's not clear yet what we're going to do about the default wallet, e.g. do we force users to give it a name? I expect only a fraction of users to care about the multi-wallet stuff, so a term like "default wallet" is confusing.

**Comment by Sjors** [comment]
tACK fe7048b

**Comment by Sjors** [comment]
Oops, now we have a trailing `-` on mainnet.

**Pull request merged**